src/wix/WixToolset.Core/Compiler.cs

```diff
@@ -5485,6 +5456,227 @@ private YesNoType ParseFileElement(XElement node, string componentId, string dir
+        /// <param name="node">File element to parse.</param>
+        /// <param name="componentId">Parent's component id.</param>
+        /// <param name="directoryId">Ancestor's directory id.</param>
+        /// <param name="diskId">Disk id inherited from parent component.</param>
+        /// <param name="sourcePath">Default source path of parent directory.</param>
+        /// <param name="possibleKeyPath">This will be set with the possible keyPath for the parent component.</param>
+        /// <param name="win64Component">true if the component is 64-bit.</param>
+        /// <param name="componentGuid">Component GUID (including `*`).</param>
+        /// <returns>Yes if this element was marked as the parent component's key path, No if explicitly marked as not being a key path, or NotSet otherwise.</returns>
+        private YesNoType ParseFileElement(XElement node, string componentId, string directoryId, int diskId, string sourcePath, out string possibleKeyPath, bool win64Component, string componentGuid)
+        {
+            var keyPath = this.ParseFileElementAttributes(node, componentId, directoryId, diskId, sourcePath, out possibleKeyPath, componentGuid, isNakedFile: false, out var fileSymbol, out var assemblySymbol);
+
+            if (!this.Core.EncounteredError)
+            {
+                this.Core.AddSymbol(fileSymbol);
+
+                if (assemblySymbol != null)
+                {
+                    this.Core.AddSymbol(assemblySymbol);
+                }
+
+                this.ParseFileElementChildren(node, fileSymbol, keyPath, win64Component);
+            }
+
+            return keyPath;
+        }
+
+        /// <summary>
+        /// Parses a file element outside a component.
+        /// </summary>
+        /// <param name="node">File element to parse.</param>
+        /// <param name="parentType">Type of complex reference parent. Will be Unknown if there is no parent.</param>
+        /// <param name="parentId">Optional identifier for primary parent.</param>
+        /// <param name="directoryId">Ancestor's directory id.</param>
+        /// <param name="sourcePath">Default source path of parent directory.</param>
+        /// <returns>Yes if this element was marked as the parent component's key path, No if explicitly marked as not being a key path, or NotSet otherwise.</returns>
+        private void ParseNakedFileElement(XElement node, ComplexReferenceParentType parentType, string parentId, string directoryId, string sourcePath)
+        {
+            var sourceLineNumbers = Preprocessor.GetSourceLineNumbers(node);
+            var win64 = this.Context.IsCurrentPlatform64Bit;
+            string condition = null;
+            string subdirectory = null;
+
+            var keyPath = this.ParseFileElementAttributes(node, "TemporaryComponentId", directoryId, diskId: CompilerConstants.IntegerNotSet, sourcePath, out var _, componentGuid: "*", isNakedFile: true, out var fileSymbol, out var assemblySymbol);
```

Comment:
How about setting this to an invalid Id like `"@WixTempComponentId"`? That way, if it ever sneaks through too far into the build process, an error will occur.

Reply:
Ooh, I like that!